src/shape.cpp

```diff
@@ -490,6 +490,26 @@ shape shape::with_lens(const std::vector<std::size_t>& l) const
     return this->with_lens(this->type(), l);
 }
 
+shape shape::prepend_dim(size_t dim) const
+{
+    if(this->dynamic())
+        MIGRAPHX_THROW("SHAPE: with_lens() called on dynamic shape");
```

Comment:
`with_lens`? Where is this being used?

Reply:
prepend_dim() is being called from matmul logic here, simply because the standard allows a one-dimensional (N) tensor multiplication -- but only after broadcasting it to two dimensions as 1 x N -- or N x 1 (for append_dim). 
(I am not sure if you are asking something else..)

Reply:
But why do you mention `with_lens()` here?

Reply:
removed changes..